gcp-common/src/main/java/io/micronaut/gcp/credentials/LogAuthenticationFailures.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.credentials;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import io.micronaut.aop.Around;
+import io.micronaut.context.annotation.Type;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation for applying authentication failure logging AOP advice to a managed instance of
+ * {@link GoogleCredentials}.
+ *
+ * @author Jeremy Grelle
+ * @since 5.2.0
+ */
+@Documented
+@Retention(RUNTIME)
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Around(proxyTargetMode = Around.ProxyTargetConstructorMode.ALLOW)
```

Comment:
Should we use? 

`@Around(proxyTargetMode = Around.ProxyTargetConstructorMode.ALLOW, proxyTarget=true)`

Reply:
I just verified via test that it seems to already be behaving as though `proxyTarget=true` in that the returned instance implements `InterceptedProxy`. Explicitly setting `proxyTarget=true` gives the same result.